### PR TITLE
registry: Avoid calling Value.setValue() when closing registry

### LINKED ISSUE #1349: protocols.http.proxy occassionally gets ignored
I have `protocols.http.proxy` set to `127.0.0.1:8118` which is Privoxy listening on 127.0.0.1 and it's configured to send traffic to Tor (`forward-socks5t / 127.0.0.1:9050 .`).

However often `web fetch https://icanhazip.com/` returns the real IP\n and `web title http://expyuzz4wqqyqhjn.onion/` (www.torproject.org, see [onion.torproject.org](https://onion.torproject.org)) returns `<urlopen error [Errno -2] Name or service not known>`.

I have confirmed that the issue is not in Privoxy configuration as `curl -x http://127.0.0.1:8118 https://icanhazip.com` returns IP of a Tor exit node.

* The current (running) version of this Limnoria is 2018.09.01, running on Python 3.6.5 (default, Apr  1 2018, 05:46:30)  [GCC 7.3.0].
* Ubuntu 18.04.1 LTS

### PR DESCRIPTION
**Pull request by BlankEclair**
Fixes #1349

### CODE REVIEW

File: src/registry.py [hunk 1]
```diff
@@ -148,7 +148,12 @@ def close(registry, filename, private=True):
                 if value._showDefault:
                     lines.append('#\n')
                     try:
-                        x = value.__class__(value._default, value._help)
+                        # We set setDefault to False and manually call
+                        # Value._setValue, just in case the class inherits
+                        # Value.setValue to set some global state (#1349)
+                        x = value.__class__(value._default, value._help,
+                            setDefault=False)
+                        x._setValue(value._default, inherited=False)
```
Comment (progval) [inline src/registry.py:156]: Why call `._setValue`?

Comment (BlankEclair) [reply src/registry.py:156]:
```
ERROR 2024-10-22T21:25:58 Exception printing default value of supybot.user:
Traceback (most recent call last):
  File "/home/claire/.local/pipx/venvs/limnoria/lib/python3.12/site-packages/supybot/registry.py", line 161, in close
    lines.append('# Default value: %s\n' % x)
                 ~~~~~~~~~~~~~~~~~~~~~~~~^~~
  File "/home/claire/.local/pipx/venvs/limnoria/lib/python3.12/site-packages/supybot/registry.py", line 684, in __str__
    s = Value.__call__(self)
        ^^^^^^^^^^^^^^^^^^^^
  File "/home/claire/.local/pipx/venvs/limnoria/lib/python3.12/site-packages/supybot/registry.py", line 568, in __call__
    return self.value
           ^^^^^^^^^^
  File "/home/claire/.local/pipx/venvs/limnoria/lib/python3.12/site-packages/supybot/registry.py", line 243, in __getattr__
    self.__nonExistentEntry(attr)
  File "/home/claire/.local/pipx/venvs/limnoria/lib/python3.12/site-packages/supybot/registry.py", line 225, in __nonExistentEntry
    raise NonExistentRegistryEntry(s)
supybot.registry.NonExistentRegistryEntry: 'value' is not a valid entry in 'unset'
```

Comment (progval) [reply src/registry.py:156]:
```suggestion
                        x.value = value._default
```

How about this? It works too, and probably spares some duplicate work (as `_setValue` is recursive); in addition to triggering whatever side-effect plugins might have in their `_setValue` implementations.

Comment (BlankEclair) [reply src/registry.py]: Done, thank you

**Comment by progval** [comment]
Thanks for the repro, troubleshooting, and the PR! That was very helpful

**Pull request merged**